script/tunnel_builder.lua

```diff
@@ -0,0 +1,212 @@
+local math2d = require('math2d')
+local util = require('util')
+local constants = require('constants')
+
+local function pairing_target(tunnel)
+	for unit,prop in pairs(global.Tunnels) do
+		if prop.player ==tunnel.index then
+			return unit
+		end
+	end
+	return nil
+end
+
+local function build_components(entity,player,name,sig_pos)
+
```

Comment:
Components are built in the same way, so there might be a way to make this into a simple for loop..

Reply:
trying to find out how to store them in an array. I do not why, but it keeps failing to store the object in array.
Possible reason is deep/shallow copy. I am keeping eyes on that to solve this.